src/api_server/swagger/firecracker.yaml

```diff
@@ -858,41 +858,47 @@ definitions:
     type: object
     required:
       - drive_id
-      - is_read_only
       - is_root_device
-      - path_on_host
     properties:
       drive_id:
         type: string
+      partuuid:
+        type: string
+        description:
+          Represents the unique id of the boot partition of this device. It is
+          optional and it will be taken into account only if the is_root_device
+          field is true.
+      is_root_device:
+        type: boolean
       cache_type:
         type: string
         description:
           Represents the caching strategy for the block device.
         enum: ["Unsafe", "Writeback"]
         default: "Unsafe"
+
+      # VirtIOBlock specific parameters
       is_read_only:
         type: boolean
-      is_root_device:
-        type: boolean
-      partuuid:
-        type: string
-        description:
-          Represents the unique id of the boot partition of this device. It is
-          optional and it will be taken into account only if the is_root_device
-          field is true.
+        description: Is block read only. This field is specific for virtio-block config.
       path_on_host:
         type: string
-        description: Host level path for the guest drive
+        description: Optional host level path for the guest drive. This field is specific for virtio-block config.
       rate_limiter:
         $ref: "#/definitions/RateLimiter"
       io_engine:
         type: string
         description:
           Type of the IO engine used by the device. "Async" is supported on
-          host kernels newer than 5.10.51.
+          host kernels newer than 5.10.51. This field is specific for virtio-block config.
         enum: ["Sync", "Async"]
         default: "Sync"
 
+      # VhostUserBlock specific parameters
+      socket:
+        type: string
+        description: Optional path to the socket of vhost-user-block backend. This field is part of vhost-user-block config.
```

Comment:
It is required for vhost-user block.
Also, let's follow the same pattern by saying "this field is specific for vhost-user-block config".

Reply:
Updated descriptions.